src/v2/controllers/vreplicaset_controller/proof/helper_lemmas.rs

```diff
@@ -128,46 +129,106 @@ pub proof fn vrs_non_interference_property_equivalent_to_lifted_vrs_non_interfer
+            }
+            assert(filtered_objs.len() == filtered_pods.len());
+            assert forall |i: int| 0 <= i < filtered_pods.len() implies filtered_pods.map_values(|p: PodView| p.marshal())[i] == filtered_objs[i] by {
+                assert(filtered_pods == resp_pods.filter(filter_pods_pred));
+                assert(PodView::unmarshal(filtered_objs[i]).is_Ok()) by {
+                    assert(forall |obj: DynamicObjectView| #[trigger] filtered_objs.contains(obj) ==> PodView::unmarshal(obj).is_Ok()) by {
+                        true_pred_on_seq_implies_true_pred_on_filtered_seq(resp_objs, |obj: DynamicObjectView| PodView::unmarshal(obj).is_Ok(), |obj: DynamicObjectView| owned_selector_match_is(vrs, obj));
+                    }
+                    true_pred_on_all_element_equal_to_pred_on_all_index(filtered_objs, |obj: DynamicObjectView| PodView::unmarshal(obj).is_Ok());
+                }
+                assert(PodView::unmarshal(filtered_objs[i]).get_Ok_0() == filtered_pods[i]);
+                unmarshal_preserves_integrity(filtered_pods[i], filtered_objs[i]);
+            }
+            assert(filtered_pods.len() == filtered_pods.map_values(|p: PodView| p.marshal()).len());
+            assert_seqs_equal!(filtered_pods.map_values(|p: PodView| p.marshal()), filtered_objs);
+        }
+    }
+    assert(filtered_pods.len() == matching_pod_entries(vrs, s.resources()).len() == matching_pods(vrs, s.resources()).len()) by {
+        assert(matching_pod_entries(vrs, s.resources()).values() == resp_objs.filter(|obj| owned_selector_match_is(vrs, obj)).to_set());
+        assert(resp_objs.no_duplicates());
+        assert(resp_objs.filter(|obj| owned_selector_match_is(vrs, obj)).no_duplicates());
+        resp_objs.filter(|obj| owned_selector_match_is(vrs, obj)).unique_seq_to_set();
+        assert_by(matching_pod_entries(vrs, s.resources()).len() == matching_pod_entries(vrs, s.resources()).values().len(), {
+            a_submap_of_a_finite_map_is_finite(matching_pod_entries(vrs, s.resources()), s.resources());
+            injective_finite_map_implies_dom_len_is_equal_to_values_len(matching_pod_entries(vrs, s.resources()));
+        });
+        assert(matching_pod_entries(vrs, s.resources()).values().len() == resp_objs.filter(|obj| owned_selector_match_is(vrs, obj)).len());
+        assert(resp_objs.filter(|obj: DynamicObjectView| owned_selector_match_is(vrs, obj)) == filtered_pods.map_values(|p: PodView| p.marshal()));
+        assert(resp_objs.filter(|obj: DynamicObjectView| owned_selector_match_is(vrs, obj)).len() == filtered_pods.map_values(|p: PodView| p.marshal()).len());
+        assert(filtered_pods.map_values(|p: PodView| p.marshal()).len() == filtered_pods.len());
+        assert(matching_pod_entries(vrs, s.resources()).len() == matching_pods(vrs, s.resources()).len());
+    }
 }
 
+#[verifier(external_body)]
```

Comment:
@marshtompsxd you may want to give your own input, but I think you said that something like this wasn't something we wanted to assume?

Reply:
@Catoverflow  I don't think we should add this assumption. Instead, we should prove that `obj == some_pod.marshal()` and since `PodView::unmarshal(obj).is_Ok() && #[trigger] PodView::unmarshal(obj).get_Ok_0() == pod` you can prove `some_pod == pod`